src/realm/list.cpp

```diff
@@ -438,6 +438,9 @@ void Lst<Mixed>::insert(size_t ndx, Mixed value)
     ensure_created();
     auto sz = size();
     CollectionBase::validate_index("insert()", ndx, sz + 1);
+    if (value.is_type(type_TypedLink)) {
```

Comment:
Aren't we missing this from the Dictionary API as well?

Reply:
There was a proper check in Dictionary already:
https://github.com/realm/realm-core/blob/fb4579c9882c7108634cc6fcbe4fd462fca30871/src/realm/dictionary.cpp#L535-L536